src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -795,6 +814,18 @@ class TreeDataSource implements IAsyncDataSource<ITreeItem, ITreeItem> {
 		}
 		return result;
 	}
+
+	async setParent(elements: ITreeItem[], newParentElement: ITreeItem): Promise<void> {
+		if (this.treeView.dataProvider) {
+			try {
+				await this.withProgress(this.treeView.dataProvider.setParent!(elements, newParentElement));
```

Comment:
Generally, we try to avoid using `!` (I know we're not perfect on this though). Can you add it to the `if` 2 lines above?